src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticListener.cs

```diff
@@ -57,7 +57,9 @@ public static IObservable<DiagnosticListener> AllListeners
         /// </summary>
         public virtual IDisposable Subscribe(IObserver<KeyValuePair<string, object>> observer, Predicate<string> isEnabled)
         {
-            return SubscribeInternal(observer, isEnabled, (name, arg1, arg2) => isEnabled(name));
+            return isEnabled == null ?
+                SubscribeInternal(observer, null, null) :
+                SubscribeInternal(observer, isEnabled, (name, arg1, arg2) => isEnabled(name));
```

Comment:
It would be more efficient to do:
```C#
if (isEnabled == null)
{
     SubscribeInternal(observer, null, null);
}
else
{
    Predicate<string> localIsEnabled = isEnabled;
    SubscribeInternal(observer, isEnabled, (name, arg1, arg2) => localIsEnabled);
}
```
Otherwise, the C# compiler will allocate the closure for the non-null branch even if isEnabled is null, as it lifts isEnabled to the closure.